sycl/test-e2e/KernelCompiler/opencl_capabilities.cpp

```diff
@@ -0,0 +1,170 @@
+//==- opencl_capabilities.cpp ----------------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: cm-compiler
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// Here we are testing some of the various args that SYCL can and cannot
+// pass to an OpenCL kernel that is compiled with the kernel_compiler.
+
+#include <sycl/sycl.hpp>
+
+namespace syclex = sycl::ext::oneapi::experimental;
+using source_kb = sycl::kernel_bundle<sycl::bundle_state::ext_oneapi_source>;
+using exe_kb = sycl::kernel_bundle<sycl::bundle_state::executable>;
+
+// -----------------------
+//  local accessor
+// -----------------------
+auto constexpr LocalAccCLSource = R"===(
+    kernel void test_la(global int *a, local float *b, int n) {
```

Comment:
The names of the variable could be improved in this test.  If I'm reading this correctly, parameter `a` is is pointer to the `b` buffer while parameter `b` is a pointer to the `acc_local` accessor.

Reply:
done